Yesod/Auth/OAuth2.hs

```diff
@@ -104,5 +108,21 @@ authOAuth2 name oauth getCreds = AuthPlugin name dispatch login
         <a href=@{tm $ oauth2Url name}>Login via #{name}
         |]
 
+-- | Handle the common case of fetching Profile information a JSON endpoint
+--
+-- Throws @'InvalidProfileResponse'@ if JSON parsing fails
+--
+handleProfileResponse :: FromJSON a
```

Comment:
I wouldn't say it does either of those things. It's already decoded when we get the `a` and `map` is very overloaded. Maybe "convert"? I chose "handle" for "error handling".